sphinx_gallery/directives.py

```diff
@@ -125,67 +127,88 @@ def run(self):
         # Retrieve source directory
         src_dir = config.sphinx_gallery_conf["src_dir"]
 
-        # Parse the argument into the individual objects
-        obj_list = []
+        # Parse the argument into the individual args
+        arg_list = []
 
         if self.arguments:
-            obj_list.extend([c.strip() for c in self.arguments[0].split()])
+            arg_list.extend([c.strip() for c in self.arguments[0].split()])
         if self.content:
-            obj_list.extend([c.strip() for c in self.content])
+            arg_list.extend([c.strip() for c in self.content])
 
         lines = []
 
         # Add a heading if requested
         if "add-heading" in self.options:
             heading = self.options["add-heading"]
             if heading == "":
-                if len(obj_list) == 1:
-                    heading = f"Examples using ``{obj_list[0]}``"
+                if len(arg_list) == 1:
+                    heading = f"Examples using ``{arg_list[0]}``"
                 else:
                     heading = "Examples using one of multiple objects"
             lines.append(heading)
             heading_level = self.options.get("heading-level", "^")
             lines.append(heading_level * len(heading))
 
-        def has_backrefs(obj):
-            path = Path(src_dir, backreferences_dir, f"{obj}.examples")
-            return path if (path.is_file() and (path.stat().st_size > 0)) else False
+        ExampleInfo = namedtuple("ExampleInfo", ["intro", "title", "arg"])
+        backreferences_all = None
+        if backreferences_dir:
+            backreferences_all = _read_json(
+                Path(src_dir, backreferences_dir, "backreferences_all.json")
+            )
 
-        file_paths = []
-        for obj in obj_list:
-            if backreferences_dir and (path := has_backrefs(obj)):
-                file_paths.append((obj, path.resolve()))
-            elif paths := Path(src_dir).glob(obj):
-                file_paths.extend([(obj, p.resolve()) for p in paths])
+        def has_backrefs(arg):
+            if backreferences_all is None:
+                return False
+            examples = backreferences_all.get(arg, None)
+            if examples:
+                return examples
+            return False
+
+        file_paths = {}
+        for arg in arg_list:
+            # Backreference arg input
+            if paths := has_backrefs(arg):
+                # Note `ExampleInfo.arg` not required for backreference paths
+                for path in paths:
+                    file_paths[Path(path[0])] = ExampleInfo(
```

Comment:
Note: `path` is the full path to the example file, from `backreferences_all.json` file, and it should by a perfectly resolved path (thus I don't use `resolve()` here) and it should match paths obtained below via `Path(src_dir).glob(arg)` (thus duplication is prevented in the dict keys)